src/Entity/Pet.php

```diff
@@ -24,46 +24,61 @@ class Pet
 
     /**
      * @ORM\Column(type="string", length=255)
-     * @Assert\NotBlank
+     * @Assert\NotBlank(groups={"new", "edit"})
      */
     private $name;
 
     /**
      * @ORM\Column(type="text")
-     * @Assert\NotBlank
+     * @Assert\NotBlank(groups={"new", "edit"})
      */
     private $description;
 
     /**
      * @ORM\Column(type="integer")
-     * @Assert\NotBlank
+     * @Assert\NotBlank(groups={"new", "edit"})
      */
     private $age;
 
     /**
      * @ORM\OneToMany(targetEntity="App\Entity\Photo", mappedBy="pet",
      *     cascade={"persist", "remove"}, orphanRemoval = true)
-     * @Assert\Valid(traverse="true")
+     * @Assert\Valid(traverse="true", groups={"new"})
      * @Assert\Count(
      *      min = "1",
      *      minMessage = "You must upload at least one image",
-     *      groups="new"
+     *      groups={"new"}
      * )
      */
     private $photos;
 
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\User", inversedBy="pets")
      * @ORM\JoinColumn(nullable=false)
+     * @Assert\NotBlank(groups={"new", "edit"})
      */
     private $owner;
 
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\Species", inversedBy="pets")
      * @ORM\JoinColumn(nullable=false)
+     * @Assert\NotBlank(groups={"new", "edit"})
```

Comment:
search related?

Reply:
Because my search form is also made based on Pet class, and in the case of search none of form fields is required, I had to define validation rules in more detail so it works for new Pet, edit Pet and search.

Reply:
and that is wrong, as I mentioned in other comment, you should have a separate class just for this